[15.05] First crack at dev release notes for 15.05 release.

jmchilton opened the pull request:
Preview - https://github.com/jmchilton/galaxy/blob/release_notes_15.05/doc/source/releases/15.05.rst

jmchilton committed:
First crack at release notes for 15.05.

martenson commented:
looks awesome to me; right on point, useful links
:+1:

jennaj commented:
Totally agree. Best version ever. 

-- jen

> On May 11, 2015, at 9:24 PM, Martin Cech notifications@github.com wrote:
> 
> looks awesome to me; right on point, useful links
> 
> —
> Reply to this email directly or view it on GitHub.

jennaj commented:
LOVE it. I think having developers put this together makes it better. Greg started these off (he was the dev PM at the time for the project). 

Saw one typo - in bugs - pull 273. 

-- jen

> On May 11, 2015, at 8:16 PM, John Chilton notifications@github.com wrote:
> 
> Preview - https://github.com/jmchilton/galaxy/blob/release_notes_15.05/doc/source/releases/15.05.rst
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/galaxyproject/galaxy/pull/253
> 
> Commit Summary
> 
> First crack at release notes for 15.05.
> File Changes
> 
> A doc/source/releases/15.05.rst (342)
> A scripts/bootstrap_history.py (82)
> Patch Links:
> 
> https://github.com/galaxyproject/galaxy/pull/253.patch
> https://github.com/galaxyproject/galaxy/pull/253.diff
> —
> Reply to this email directly or view it on GitHub.

jennaj commented:
pull 231 - sorry! Also meant to send that just you, John. Opps! Is late.. 

-- jen

> On May 11, 2015, at 10:19 PM, Jennifer Jackson jenhjackson@gmail.com wrote:
> 
> LOVE it. I think having developers put this together makes it better. Greg started these off (he was the dev PM at the time for the project). 
> 
> Saw one typo - in bugs - pull 273. 
> 
> -- jen
> 
> > On May 11, 2015, at 8:16 PM, John Chilton notifications@github.com wrote:
> > 
> > Preview - https://github.com/jmchilton/galaxy/blob/release_notes_15.05/doc/source/releases/15.05.rst
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> >   https://github.com/galaxyproject/galaxy/pull/253
> > 
> > Commit Summary
> > 
> > First crack at release notes for 15.05.
> > File Changes
> > 
> > A doc/source/releases/15.05.rst (342)
> > A scripts/bootstrap_history.py (82)
> > Patch Links:
> > 
> > https://github.com/galaxyproject/galaxy/pull/253.patch
> > https://github.com/galaxyproject/galaxy/pull/253.diff
> > —
> > Reply to this email directly or view it on GitHub.

jmchilton committed:
Add more links to 15.05 release notes - Trello and otherwise.

jmchilton commented:
Thanks for the comments @martenson and @jennaj.

About the PR 231 - I think the implied typo was "collectl" - which agree looks a lot like a typo but is actually the name of the software. I have updated the PR to add a link to it to clarify this.

I have also added links to all relevant Trello cards that I could find (unless the card was only an auto-generated link back to the PR) and there weren't many.

natefoo commented:
:thumbsup::thumbsup::thumbsup::thumbsup::thumbsup::thumbsup::thumbsup::thumbsup:

dannon commented:
+1, huge fan of docs living with the source.

nsoranzo commented on doc/source/releases/15.05.rst:
```diff
@@ -0,0 +1,348 @@
+* Pluggable framework to custom authentication (including new LDAP/Active
+  Directory integration). Thanks to many including Andrew Robinson,
+  Nicola Soranzo, and David Trudgian. `Pull Request 1`_, `Pull Request 33`_,
+  `Pull Request 51`_, `Pull Request 74`_, `Pull Request 98`_,
```
s/74/75/

nsoranzo commented on doc/source/releases/15.05.rst:
```diff
@@ -0,0 +1,348 @@
+  Nicola Soranzo, and David Trudgian. `Pull Request 1`_, `Pull Request 33`_,
+  `Pull Request 51`_, `Pull Request 74`_, `Pull Request 98`_,
+  `Pull Request 216`_
+* Implement a new ``section`` tag for tools. `Pull Request 35`_,
```
s/tools/tool parameters/

Otherwise it may seem you're speaking of the tool menu in the left pane.

jmchilton committed:
Suggestions for release notes 15.05 from @nsoranzo.

nsoranzo commented on doc/source/releases/15.05.rst:
```diff
@@ -0,0 +1,348 @@
+* Remove demo sequencer app. 3af3bf5_
+* Tweaks to the Pulsar's handling of async messages. `Pull Request 109`_
+* Return more specific API authentication errors. 71a64ca_
+* Upgrade Python dependency sqlalchemy to 1.0.0. d725aab_
```
Pull Request 129.

nsoranzo commented on doc/source/releases/15.05.rst:
```diff
@@ -0,0 +1,348 @@
+* Return more specific API authentication errors. 71a64ca_
+* Upgrade Python dependency sqlalchemy to 1.0.0. d725aab_
+* Upgrade Python dependency amqp to 1.4.6. e09761e_
+* Upgrade Python dependency kombu to 3.0.24. 8d3c531_
```
amqp and kombu are Pull Request 128.

nsoranzo commented on doc/source/releases/15.05.rst:
```diff
@@ -0,0 +1,348 @@
+* Fix to suppress pysam binary incompatibility warning when using datatypes
+  in ``binary.py``. `Pull Request 252`_
+* Allow a tool data table to declare that duplicate entries are not
+  allowed. `Pull Request 245`_
```
This looks more like an Enhancement than a Fix.

jmchilton replied:
Yes, yes it does. Though somehow it was merged after the deadline for new features :).

I'll move this.

nsoranzo commented:
Awesome work, thanks @jmchilton!

jmchilton committed:
More updates to release 15.05 notes based on @nsoranzo's comments.

jmchilton commented:
Thanks for the detailed review @nsoranzo - I believe I have updated it to reflect all of your comments.

martenson committed:
add some trello card links

jmchilton committed:
Merge pull request #2 from martenson/patch-1

add some trello card links to release notes

nsoranzo commented:
Thanks! I think that you may drop the commit ids where you added the PR number, but not really an issue! :+1:

bgruening commented:
Should we add this as well?

```
* Add argument to tool parameters. `Pull Request 8`_

.. _Pull Request 8: https://github.com/galaxyproject/galaxy/pull/8

```

jmchilton commented:
Absolutely @bgruening - I don't know how I missed that. Will update.

bgruening commented:
You guys are implementing features faster than we can document them :)

jmchilton committed:
Add PR7 to release notes (caught by @bgruening).

... small updates to bootstrap history script for reuse in planemo.

blankenberg committed:
all data source tools go to _top

jmchilton committed:
Merge pull request #3 from blankenberg/patch-1

all data source tools go to _top

carlfeberhard commented:
+1 Thanks, John

jmchilton commented:
After more conversations with @dannon, @jennaj, and @martenson  - I think the current consensus is to:
- Include typical announcements in the source tree as RST also - information on getting galaxy and highlights for the release (as well as embedding these release notes as part of that).
- Not to duplicate this on the wiki and link to either Github or RTD instead (RTD I guess since github doesn't allow embedding RST).
- RTD will be the canonical link for release announcement instead of wiki (so other announcements like biostar and twitter should link to that).
- Once the announcement is published - adding just a link on the wiki for RSS/consistency....

Objections?

natefoo commented:
All in favor... Aye.

dannon commented:
Yep, consensus is accurate.  Let's do it.

On Wed, May 13, 2015 at 9:24 AM Nate Coraor notifications@github.com
wrote:

> All in favor... Aye.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/galaxyproject/galaxy/pull/253#issuecomment-101662598.

jmchilton committed:
WIP: Experiment porting release announcement to rST/RTD.

jmchilton committed:
WIP: Announcement updates.

jmchilton commented:
Seeking input on possible announcement layout:
http://jmchilton-galaxy.readthedocs.org/en/latest/releases/15.05_announce.html

Questions:
- Do we really want RTDs advertisements on our release announcements?
- I cannot get the image to float right like on the wiki on RTD (I can with a direct sphinx build on local machine - but RTD looses the styling).
- Any other advice?

Given the way it turned out - I wonder if we should just stick to the wiki for this stuff and link to the release notes.

martenson commented:
+1 to stick to wiki for now, (probably until wiki rebirth)
I am starting to  modify https://wiki.galaxyproject.org/News/2015_05_GalaxyRelease

natefoo commented:
I still like the RTD. And the ads are not very intrusive.

jxtx commented:
I also like RTD and I don't have an issue with the advertisement (
acknowledgement of sponsorship).

-- jt

On Wed, May 13, 2015 at 12:35 PM, Nate Coraor notifications@github.com
wrote:

> I still like the RTD. And the ads are not very intrusive.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/galaxyproject/galaxy/pull/253#issuecomment-101739417.

martenson committed:
update logo

martenson committed:
link to wiki SourceCode instead of getgalaxy.org

jmchilton committed:
Merge pull request #5 from martenson/patch-3

link to wiki SourceCode instead of getgalaxy.org

martenson committed:
shrink to 25%; switch to projectless version

martenson commented:
Some logo discussion here: https://github.com/jmchilton/galaxy/pull/4

jmchilton committed:
Merge pull request #4 from martenson/patch-2

update logo

jmchilton commented:
Okay - if even @jxtx has no problem with the "ad" - I am happy to throw my :+1: to using RTD.

martenson commented:
+1 to RTD, I will have one more PR, please don't merge yet

dannon commented:
+1

jmchilton commented:
@jennaj are you okay with RTD then (Would look something like - http://jmchilton-galaxy.readthedocs.org/en/latest/releases/15.05_announce.html - but @martenson has modified the logo).

Switching to content discussions then - @natefoo did you pick these `git` install instructions? I feel like we have two different ways to update listed and neither are the one I would recommend using. If I am reading these specific notes - chances are I would want to update to `release_15.05` right? I feel like I would include just that command and not the other two.

martenson committed:
add some ads

jmchilton committed:
Merge pull request #6 from martenson/patch-4

add some ads

martenson commented:
@jmchilton Also the hg commands probably only work if you are tracking -dist (central is abandoned, right?). Worth mentioning?

martenson commented:
![screenshot 2015-05-13 13 41 22](https://cloud.githubusercontent.com/assets/1814954/7616879/ca96e950-f975-11e4-9faa-4f51968374d7.png)

natefoo commented:
The hg commands are fine, @martenson. And galaxy-central is not abandoned, although it does not differ from galaxy-dist.

jmchilton committed:
Refactor thanks from annoucements out.

martenson commented:
@natefoo my bad, I thought -central won't be updated from now on.

jmchilton commented:
@martenson RTD caught up - the logo looks a little off since it is not transparent. http://jmchilton-galaxy.readthedocs.org/en/latest/releases/15.05_announce.html

Any chance there is a transparent version?

martenson commented:
on it

On Wed, May 13, 2015 at 1:51 PM John Chilton notifications@github.com
wrote:

> @martenson https://github.com/martenson RTD caught up - the logo looks
> a little off since it is not transparent.
> http://jmchilton-galaxy.readthedocs.org/en/latest/releases/15.05_announce.html
> 
> Any chance there is a transparent version?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/galaxyproject/galaxy/pull/253#issuecomment-101758246.

dannon commented:
Can we use the .svg for the logo?  Looks a bit blurry and pixellated on my screen.

martenson commented:
@dannon I installed Inkscape but making transparent .svg is AMP. I can try reducing the compression. Or you can try yourself using the pdf we have in the shared Galaxy folder.

jmchilton committed:
Fill out doc stubs for releases through 13.01.

These pages have the header and footer and links to the relevant pages on the wiki.

These are the ones on github and the ones that got at least a couple of the most serious bug fixes announced over the last year.

dannon commented:
Ahh, k, from the pdf.  Do we not have source files for the logo?

martenson committed:
update to transparent logo

jmchilton commented:
Stubbed out links for past releases as well for people navigating around on RTD - preview @ http://jmchilton-galaxy.readthedocs.org/en/latest/releases/index.html.

jmchilton committed:
Add missing release to doc index.

natefoo committed:
Add a stub for older release announcements.

martenson committed:
Merge pull request #7 from natefoo/release_notes_15.05

Add a stub for older release announcements
